src/Operation/CollMod.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/*
+ * Copyright 2018 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace MongoDB\Operation;
+
+use MongoDB\Driver\Command;
+use MongoDB\Driver\Server;
+use MongoDB\Driver\Exception\RuntimeException as DriverRuntimeException;
+use MongoDB\Exception\InvalidArgumentException;
+use MongoDB\Exception\UnexpectedValueException;
+
+/**
+ * Operation for the collMod command.
+ *
+ * @api
+ * @see \MongoDB\Collection::collMod()
+ * @see http://docs.mongodb.org/manual/reference/command/collMod/
+ */
+class CollMod implements Executable
+{
+    private $databaseName;
+    private $collectionName;
+    private $collModOption;
+    private $collModValue;
+    private $options;
+
+    /**
+     * Constructs a collMod command.
+     *
+     * Supported options:
+     *
+     *  * typeMap (array): Type map for BSON deserialization. This will only be
+     *    used for the returned command result document.
+     *
+     * @param string       $databaseName     Database name
+     * @param string       $collectionOrView Collection or view to modify
+     * @param array|object $collModOption    Option for collMod
+     * @param array        $collModValue     Value for collModOption
+     * @param array        $options          Command options
+     * @throws InvalidArgumentException for parameter/option parsing errors
+     */
+    public function __construct($databaseName, $collectionOrView, $collModOption, $collModValue, array $options = [])
+    {
+        if (isset($options['typeMap']) && ! is_array($options['typeMap'])) {
+            throw InvalidArgumentException::invalidType('"typeMap" option', $options['typeMap'], 'array');
+        }
+
+        $this->databaseName = (string) $databaseName;
+        $this->collectionOrView = (string) $collectionOrView;
+        $this->collModOption = $collModOption;
+        $this->collModValue = $collModValue;
+        $this->options = $options;
+    }
+
+    /**
+     * Execute the operation.
+     *
+     * @see Executable::execute()
+     * @param Server $server
+     * @return array|object Command result document
+     * @throws UnexpectedValueException if the command response was malformed
+     * @throws DriverRuntimeException for other driver errors (e.g. connection errors)
+     */
+    public function execute(Server $server)
+    {
+        $cursor = $server->executeCommand($this->databaseName, $this->createCommand());
```

Comment:
I think we can use `executeWriteCommand()` here.

Note that CreateCollection also supports a `writeConcern` option. Both `create` and `collMod` are listed in [commands that write](https://github.com/mongodb/specifications/blob/master/source/read-write-concern/read-write-concern.rst#other-commands-that-write) in the write concern spec.

Note: that option isn't mentioned in the MongoDB manual but there's an open ticket to do so ([DOCS-9822](https://jira.mongodb.org/browse/DOCS-9822)).